### PR TITLE
 Continue ByteArrayToString and StringToByteArray support

### PR DESCRIPTION
**Pull request by ggumen**
Addressing feedback from https://github.com/ni/VireoSDK/pull/432/ 

- Updating 3rd party library to use the latest from sourceforge.
- Adding test cases for invalid encoding type
- Not error check StringToByteArray, but keep error checking on ByteArrayToString
- minor code review feedback

### CODE REVIEW

**Comment by rajsite** [comment]
Dang it. I looked at the source forge again and found this: https://sourceforge.net/p/utfcpp/news/2015/09/moving-to-github/

The actual official utf8 repo with the latest changes is here: https://github.com/nemtrif/utfcpp

And this is the latest release: https://github.com/nemtrif/utfcpp/releases/tag/v2.3.5

The file structure looks the same so it should be really easy to upgrade to the latest release. I would be okay just taking the source folder from the release and keeping the structure of the source folder the README.md (ie we dont need to check-in the test infrastructure)

**Comment by rajsite** [comment]
To keep the 2.3.5 README.md we may update the structure to something like:
```
/include/thirdparty/utf8/<files from release>
/include/thirdparty/utf8.README.md <-- which is our README that points to the other utf8 source GitHub
```

File: source/core/NumericString.cpp [hunk 1]
```diff
@@ -2715,6 +2800,8 @@ DEFINE_VIREO_BEGIN(NumericString)
     DEFINE_VIREO_FUNCTION(StringFormat, "p(i(VarArgCount) o(String)   i(String) io(ErrorCluster err) i(StaticTypeAndData))")
     DEFINE_VIREO_FUNCTION(StringScanValue, "p(i(String) o(String) i(String) o(StaticTypeAndData))")
     DEFINE_VIREO_FUNCTION(StringScan, "p(i(VarArgCount) i(String) o(String) i(String) i(UInt32) o(UInt32) io(ErrorCluster err) o(StaticTypeAndData))")
+    DEFINE_VIREO_FUNCTION(ByteArrayToString, "p(o(String) i(Array) i(Int32) io(ErrorCluster err))")
```
Comment (cglzaguilar) [inline source/core/NumericString.cpp]: Please update the order of parameters, inputs first, outputs last, and also update the param block declaration

DEFINE_VIREO_FUNCTION(ByteArrayToString, "p(i(Array) i(Int32) o(String) io(ErrorCluster err))")
DEFINE_VIREO_FUNCTION(StringToByteArray, "p(i(String) i(Int32) o(Array) io(ErrorCluster err))")

**Comment by ggumen** [comment]
Sankara is continuing his Pull Request. Abandoning this one.

**Pull request closed without merging**